hack/setup-kind-in-ci.sh

```diff
@@ -0,0 +1,149 @@
+HELP
+}
+
+# process command line arguments
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -dorp|--docker-or-podman)     DORP="$2";                  shift;shift; ;;
+    -h|--help)                    helpmsg;                    exit 1       ;;
+    -iv|--istio-version)          ISTIO_VERSION="$2";         shift;shift; ;;
+    *) echo "Unknown argument: [$key]. Aborting."; helpmsg; exit 1 ;;
+  esac
+done
+
+# abort on any error
+set -e
+
+# set up some of our defaults
+DORP="${DORP:-docker}"
+
+# Only if this is set to "true" will Istio be installed if it is missing
+INSTALL_ISTIO="${INSTALL_ISTIO:-true}"
+
+# print out our settings for debug purposes
+cat <<EOM
+=== SETTINGS ===
+DORP=$DORP
+ISTIO_VERSION=$ISTIO_VERSION
+=== SETTINGS ===
+EOM
+
+infomsg "Make sure everything exists"
+which kubectl > /dev/null || (infomsg "kubectl executable is missing"; exit 1)
+which kind > /dev/null || (infomsg "kind executable is missing"; exit 1)
+which "${DORP}" > /dev/null || (infomsg "[$DORP] is not in the PATH"; exit 1)
+
+infomsg "Kind cluster to be created with name [ci]"
+cat <<EOF | kind create cluster --name ci --config -
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+  - role: worker
+EOF
+
+# When a new cluster is created, kind automagically sets the kube context.
+
+infomsg "Create Kind LoadBalancer via MetalLB"
+lb_addr_range="255.70-255.84"
+
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.12.1/manifests/namespace.yaml
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.12.1/manifests/metallb.yaml
+
+subnet=$(${DORP} network inspect kind --format '{{(index .IPAM.Config 0).Subnet}}')
+subnet_trimmed=$(echo "${subnet}" | sed -E 's/([0-9]+\.[0-9]+)\.[0-9]+\..*/\1/')
+first_ip="${subnet_trimmed}.$(echo "${lb_addr_range}" | cut -d '-' -f 1)"
+last_ip="${subnet_trimmed}.$(echo "${lb_addr_range}" | cut -d '-' -f 2)"
+cat <<LBCONFIGMAP | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    address-pools:
+    - name: default
+      protocol: layer2
+      addresses: ['${first_ip}-${last_ip}']
+LBCONFIGMAP
+
+# if [ -n "${ISTIO_VERSION}" ]; then
+# DOWNLOAD_ISTIO_VERSION_ARG="--istio-version ${ISTIO_VERSION}"
+# fi
+
+infomsg "Downloading istio"
+hack/istio/download-istio.sh
+
+infomsg "Installing istio"
+hack/istio/install-istio-via-istioctl.sh --client-exe-path "$(which kubectl)"
+  
+infomsg "Pushing the images into the cluster..."
+make -e DORP="${DORP}" -e CLUSTER_TYPE="kind" -e KIND_NAME="ci" cluster-push-kiali
+
+infomsg "Installing kiali server via Helm"
+helm install \
+  --namespace istio-system \
+  --set auth.strategy="anonymous" \
+  --set deployment.service_type="LoadBalancer" \
+  --set deployment.image_name=kiali/kiali \
+  --set deployment.image_version=dev \
+  --set deployment.image_pull_policy="Never" \
+  --repo https://kiali.org/helm-charts \
```

Comment:
Using the --repo option with the kiali.org chart location is most likely going to be a problem. Because this is going to install using the last released helm chart. But it is commonly the case that we change the helm charts throughout a sprint (adding permissions, adding/changing configuration settings, changing Deployment or other resource yaml templates, etc). So we will want to test the master build of the helm charts, otherwise, this could fail because it will be installing with the helm chart that may not be compatible with the master branch of kiali/kiali.

To do this, we would need to checkout the master branch of the helm chart repo, run `make build-helm-charts` and then use the helm charts in the helm chart repo's `_output` directory. The helm command would look like:

```
helm install \
  ... the --set and --namespace options ... \
  kiali-server \
 _output/charts/kiali-server-*-SNAPSHOT.tgz
```

Reply:
hmmmm wouldn't we need to merge the changes into the helm chart before running the tests here then? Otherwise we'd need to target the branch with the helm changes to coincide with the kiali PR. Same thing I was describing here: https://github.com/kiali/kiali/issues/4833#issuecomment-1075605397.

Reply:
Yes. We would need to make sure any helm chart PR dependencies are merged first.